db/document.go

```diff
@@ -332,6 +333,21 @@ func (doc *Document) GetMetaMap(userXattrKey string) (map[string]interface{}, er
 	}, nil
 }
 
+func (doc *Document) SetCrc32cUserXattrHash(userXattrKey string) {
+	if len(doc.rawUserXattr) > 0 {
+		doc.SyncData.Crc32cUserXattr = base.Crc32cHashString(doc.rawUserXattr)
+		return
+	}
+
+	// If userXattrKey is still set and there is no rawUserXattr then the xattr has been removed from the doc and we can
+	// set to empty hash.
+	// If userXattr key is no longer set but the crc32 hash is still set then we can set it to empty hash. This likely
+	// means the feature was enabled but later disabled.
+	if len(doc.rawUserXattr) == 0 || userXattrKey == "" {
+		doc.SyncData.Crc32cUserXattr = ""
+	}
```

Comment:
The previous suggestion applied to the whole function, there's no need for another length check if we move the "success" path out and to the bottom, and make the `= ""` case exit.

Reply:
Yeah makes sense. Sorry flipping between branches and missing things.